### PR TITLE
fix ci failure

### PR DESCRIPTION
**Pull request by mingwli**
Fix [test failure](https://dev.azure.com/ceapex/Engineering/_build/results?buildId=456486&view=logs&j=2b859c7d-aaae-547b-0354-c90791ad9984&t=1beb5644-7360-5729-b1db-6db8eb6a2b48)
-  separate build step for easier identifying build error
- fix IDExxx warnings
- remove docfx dependency in regression-test

### CODE REVIEW

**Comment by mingwli** [comment]
It should be dotnet sdk version issue based on the passed build: https://dev.azure.com/ceapex/Engineering/_build/results?buildId=457126&view=results
which use 5.0.203

**Comment by yufeih** [comment]
@AngryBerryMS , it is possible to upgrade the NuGet.exe in 1ES hosted agent to the latest (5.9.1) ?

**Comment by yufeih** [comment]
@mingwli , this should be caused by these IDE errors.

```
error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1789762Z D:\a\_work\1\s\src\docfx\Errors.cs(347,24): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1790503Z D:\a\_work\1\s\src\docfx\Errors.cs(688,24): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1791254Z D:\a\_work\1\s\src\docfx\Errors.cs(354,24): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1792150Z D:\a\_work\1\s\src\docfx\Errors.cs(361,24): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1793003Z D:\a\_work\1\s\src\docfx\Errors.cs(696,24): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1793749Z D:\a\_work\1\s\src\docfx\Errors.cs(414,24): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1794586Z D:\a\_work\1\s\src\docfx\Errors.cs(449,24): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1795522Z D:\a\_work\1\s\src\docfx\Errors.cs(421,24): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1796679Z D:\a\_work\1\s\src\docfx\Errors.cs(424,23): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1797668Z D:\a\_work\1\s\src\docfx\Errors.cs(699,24): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
2021-05-28T00:54:00.1798406Z D:\a\_work\1\s\src\docfx\Errors.cs(457,24): error IDE0090: 'new' expression can be simplified [D:\a\_work\1\s\src\docfx\docfx.csproj]
```

File: azure-pipelines-v3.yml [hunk 1]
```diff
@@ -31,13 +31,23 @@ jobs:
...
+
+
   # dotnet test
   - task: DotNetCoreCLI@2
     displayName: dotnet test
     condition: and(succeeded(), ne(variables['warm-up'], 'true'))
     inputs:
       command: test
-      arguments:  -warnAsError -c Release /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura
+      arguments:  -warnAsError -c Release --no-restore /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura
```
Comment (yufeih) [inline azure-pipelines-v3.yml:50]: Looks like we need an additional `dotnet build -warnAsError -c Release` step to tell between build failure and test failure.

**Review by yufeih — APPROVED** [review]
👍

**Review by live1206 — APPROVED** [review]

**Pull request merged**